cpp/src/aztec/honk/pcs/gemini/gemini.test.cpp

```diff
@@ -1,18 +1,104 @@
 #include "gemini.hpp"
 
 #include "../commitment_key.test.hpp"
+#include "polynomials/polynomial.hpp"
+#include <cstddef>
 #include <gtest/gtest.h>
+#include <gtest/internal/gtest-type-util.h>
+#include <span>
 
 namespace honk::pcs::gemini {
 
-template <class Params> class GeminiTest : public CommitmentTest<Params> {};
+template <class Params> class GeminiTest : public CommitmentTest<Params> {
+    using Gemini = MultilinearReductionScheme<Params>;
+    using Fr = typename Params::Fr;
+    using Commitment = typename Params::Commitment;
+    using Polynomial = typename barretenberg::Polynomial<Fr>;
+
+  public:
+    void execute_gemini_and_verify_claims(size_t log_n,
+                                          std::vector<Fr> multilinear_evaluation_point,
+                                          std::vector<Fr> multilinear_evaluations,
+                                          std::vector<std::span<Fr>> multilinear_polynomials,
+                                          std::vector<std::span<Fr>> multilinear_polynomials_to_be_shifted,
+                                          std::vector<Commitment> multilinear_commitments,
+                                          std::vector<Commitment> multilinear_commitments_to_be_shifted)
+    {
```

Comment:
Good idea to make a general function for testing. It makes it a lot easier to see what test cases we are actually running!